### PR TITLE
AB#2857 new home address validation

### PR DESCRIPTION
**Pull request by leiye29**
### Description

1. add validation into home-address/edit
2. map address api to application address when get address from api
3. map application address to address api when try to save address

### Checklist

- [x] I have tested the changes locally
- [ ] I have updated the documentation if necessary
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that my code follows the project's coding style by running `npm run format:check`
- [x] I have checked that my code contains no linting errors by running `npm run lint`
- [x] I have checked that my code contains no type errors by running `npm run typecheck`
- [x] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [x] I have checked that all e2e tests pass by running `npm run test:e2e -- run`

### CODE REVIEW

**Review by sebastien-comeau — CHANGES_REQUESTED** [review]

File: frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx [hunk 1]
```diff
@@ -78,39 +80,63 @@ export async function action({ request }: ActionFunctionArgs) {
...
+  const errorSummaryId = 'error-summary';
+
+  const defaultValues = actionData?.formData ??
+    addressInfo ?? {
+      address: ' ',
+      city: ' ',
+      province: ' ',
+      country: ' ',
+      postalCode: ' ',
+    };
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx]: fixed

**Review by nicholas-ly — DISMISSED** [review]

File: frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx [hunk 2]
```diff
@@ -78,39 +80,63 @@ export async function action({ request }: ActionFunctionArgs) {
...
+
+    /**
+     * The 'as any' is employed to circumvent typechecking, as the type of
+     * 'errorI18nKey' is a string, and the string literal cannot undergo validation.
+     */
+    return t(`personal-information:home-address.edit.error-message.${errorI18nKey}` as any);
+  }
+
+  const errorMessages = {
+    form: getErrorMessage(actionData?.errors.formErrors?.[0]),
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.edit.tsx]: removed.

**Review by nicholas-ly — DISMISSED** [review]

File: frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx [hunk 3]
```diff
@@ -54,29 +54,25 @@ export default function ConfirmAddress() {
...
-                {homeAddressInfo?.addressApartmentUnitNumber} {homeAddressInfo?.addressStreet}
+                {homeAddressInfo?.city}, {homeAddressInfo?.province ? homeAddressInfo?.province + ', ' : ''}
+                {homeAddressInfo?.country}
               </p>
-              <p className="m-0">
-                {homeAddressInfo?.addressCity}, {homeAddressInfo?.addressProvince ? homeAddressInfo?.addressProvince + ', ' : ''}
-                {homeAddressInfo?.addressCountry}
-              </p>
-              <p className="m-0">{homeAddressInfo?.addressPostalZipCode}</p>
+              <p className="m-0">{homeAddressInfo?.postalCode}</p>
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx]: updated

File: frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx [hunk 4]
```diff
@@ -54,29 +54,25 @@ export default function ConfirmAddress() {
...
-                {newHomeAddress.addressApartmentUnitNumber} {newHomeAddress.addressStreet}
-              </p>
-              <p className="m-0">
-                {newHomeAddress.addressCity}, {newHomeAddress.addressProvince ? newHomeAddress.addressProvince + ', ' : ''}
-                {newHomeAddress.addressCountry}
+                {newHomeAddress.city}, {newHomeAddress.province ? newHomeAddress.province + ', ' : ''}
+                {newHomeAddress.country}
               </p>
-              <p className="m-0">{newHomeAddress.addressPostalZipCode}</p>
+              <p className="m-0">{newHomeAddress.postalCode}</p>
```
[the remark this answers is not in the document]
Comment (leiye29) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx]: updated

[the remark this answers is not in the document]
Comment (nicholas-ly) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx]: Resolving as `Address` component is now used.

File: frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nicholas-ly) [reply frontend/app/routes/_gcweb-app.personal-information.home-address.confirm.tsx]: Resolving as `Address` component is now used.

**Review by nicholas-ly — APPROVED** [review]

**Pull request merged**